lasagne/layers/pool.py

```diff
@@ -18,18 +17,18 @@
 class MaxPool2DLayer(Layer):
     def __init__(self, incoming, ds, ignore_border=False, **kwargs):
         super(MaxPool2DLayer, self).__init__(incoming, **kwargs)
-        self.ds = ds # a tuple
+        self.ds = ds  # a tuple
         self.ignore_border = ignore_border
 
     def get_output_shape_for(self, input_shape):
-        output_shape = list(input_shape) # copy / convert to mutable list
+        output_shape = list(input_shape)  # copy / convert to mutable list
 
         if self.ignore_border:
-            output_shape[2] = int(np.floor(float(output_shape[2]) / self.ds[0]))
-            output_shape[3] = int(np.floor(float(output_shape[3]) / self.ds[1]))
+            output_shape[2] = int(np.floor(float(output_shape[2])/self.ds[0]))
+            output_shape[3] = int(np.floor(float(output_shape[3])/self.ds[1]))
         else:
-            output_shape[2] = int(np.ceil(float(output_shape[2]) / self.ds[0]))
-            output_shape[3] = int(np.ceil(float(output_shape[3]) / self.ds[1]))
+            output_shape[2] = int(np.ceil(float(output_shape[2])/self.ds[0]))
+            output_shape[3] = int(np.ceil(float(output_shape[3])/self.ds[1]))
```

Comment:
> Because we need to round up, not down.

Oops, didn't see the difference between the two cases at first glance. Haven't slept enough. Consistency is good then!